src/helpers.js

```diff
@@ -0,0 +1,40 @@
+
+const BASE_URL = 'https://www.omdbapi.com/?apikey=b8cd40cf&s=';
+
+export const getMoviesFromServer = async(
+  title,
+  setTitle,
+  handleAdd,
+  handleError,
+) => {
+  const response = await fetch(`${BASE_URL}${title}`);
+
+  if (!response.ok) {
+    return false;
+  }
+
+  const resultObject = await response.json();
+
+  if (resultObject.Error) {
+    handleError(resultObject.Error);
+
+    return false;
+  }
+
+  const moviesFromServer = await resultObject.Search;
+  const moviesList = await Promise.all(moviesFromServer.map(movie => (
+    fetch(`https://www.omdbapi.com/?apikey=b8cd40cf&i=${movie.imdbID}`)
+      .then(r => r.json()))));
```

Comment:
```suggestion
  const moviesList = await Promise.all(
    moviesFromServer.map(movie => (
      fetch(`https://www.omdbapi.com/?apikey=b8cd40cf&i=${movie.imdbID}`)
        .then(r => r.json())
    )),
  );
```
Так, по-моему, намного понятнее какая скобочка за что отвечает